core/src/main/java/org/elasticsearch/index/query/AndQueryParser.java

```diff
@@ -72,21 +72,11 @@ public Query parse(QueryParseContext parseContext) throws IOException, QueryPars
                 } else if (parseContext.isDeprecatedSetting(currentFieldName)) {
                     // skip
                 } else if (token == XContentParser.Token.START_ARRAY) {
-                    if ("filters".equals(currentFieldName)) {
-                        queriesFound = true;
-                        while ((token = parser.nextToken()) != XContentParser.Token.END_ARRAY) {
-                            Query filter = parseContext.parseInnerFilter();
-                            if (filter != null) {
-                                queries.add(filter);
-                            }
-                        }
-                    } else {
-                        queriesFound = true;
-                        while ((token = parser.nextToken()) != XContentParser.Token.END_ARRAY) {
-                            Query filter = parseContext.parseInnerFilter();
-                            if (filter != null) {
-                                queries.add(filter);
-                            }
+                    queriesFound = true;
+                    while ((token = parser.nextToken()) != XContentParser.Token.END_ARRAY) {
+                        Query filter = parseContext.parseInnerFilter();
+                        if (filter != null) {
+                            queries.add(filter);
```

Comment:
Had another quick look at the parser class and realized that top level array is already parsed if the first token in the and-query is start of array. In this case I think we can delete the `else` branch so when array is specified it either needs to be top level or be nested in a filters element. Otherwise no filters are parsed and we later throw QueryParsingException here.